### DISCUSSION TITLE
How can a RealVectorValue be passed to cleavage_plane_normal in the ACInterfaceCleavageFracture

### DISCUSSION THREAD

**Discussion by LifengGan** [idaholab/moose#28022] [Q&A Modules: Phase field] [answered]
Dear all moose users,

How can a `RealVectorValue` be passed to `cleavage_plane_normal` in the `ACInterfaceCleavageFracture`?

I am calculating the cleavage_plane_normal in a material properties as RealVectorValue, which is the `stress_direction`

then constructed  this in `[AuxKernels]` as a AuxVariables
```
[./direction_1]
    type = MaterialRealVectorValueAux
    property = stress_direction
    variable = direction_1
    component = 0
    execute_on = TIMESTEP_END
  [../]
  [./direction_2]
    type = MaterialRealVectorValueAux
    property = stress_direction
    variable = direction_2
    component = 1
    execute_on = TIMESTEP_END
  [../]
  [./direction_3]
    type = MaterialRealVectorValueAux
    property = stress_direction
    variable = direction_3
    component = 2
    execute_on = TIMESTEP_END
  [../]
  [./build_vector]
    type = BuildArrayVariableAux
    variable = vector
    component_variables = 'direction_1 direction_2 direction_3'
  [../]
```
 but  I got error in the step of passing the value into `cleavage_plane_normal`
" connot convert filed 'Kernels/ACInterfaceCleavageFracture/cleavage_plane_normal' value `vector` to float"
```
[./ACInterfaceCleavageFracture]
    type = ACInterfaceCleavageFracture
    variable = c
    beta_penalty = 1
    cleavage_plane_normal = vector
 [../]
```

How can I make this come true?

Best regards,
Lifeng

**Comment by GiudGiud** [comment]
Hello

It expects a single constant value:
```
  params.addRequiredParam<RealVectorValue>("cleavage_plane_normal",
                                           "Normal to the weak cleavage plane");
```
it does not expect a vector variable.
If you want to use a vector variable, likely because you want a spatial dependence, you will need to create a copy of this kernel, with a different name, and change the code a little bit to use:
```
params.addCoupledVar("cleavage_plane_normal", "description");
```
for the parameter
and then use the vector routines from the Coupleable API

Guillaume

**Reply by LifengGan** [reply]
hello, 

What if i just want it simple?

it expects in form of  `cleavage_plane_normal = '0.707 0.707 0'`

but I cannot make it with `cleavage_plane_normal  = 'direction_1 direction_2 direction_3'`

where the `direction_i` is output from `MaterialRealVectorValueAux` using `component = i`

Lifeng

**Reply by GiudGiud** [reply]
no you cant do that without changing the code.
Replacing the constant vector with a material property is the same amount of effort as using a field variable

**Reply by LifengGan** [reply, marked as answer]
Thanks for the answer
 
it works by changing the code `ACInterfaceCleavageFracture` with
`_cleavage_plane_normal(getMaterialPropertyByName<RealVectorValue>("cleavage_plane_normal"))`

Lifeng
